concepts/option/about.md

```diff
@@ -1 +1,21 @@
-TODO: add information on option concept
+The `Option` type is an enum provided by the standard library. It is useful for expressing whether a value exists or does not exist.
+While this may sound like `null` or `nil`, it is quite different and special.
+`Option` is distinctive to Rust. It is also quite fun to work with!
+
+`Option<T>` holds other types and has two variants `Some` and `None`.
+
+For example, `Option<u32>` holds the `u32` type.
+
+The below snippets demonstrate what this would look like in some Rust code.
```

Comment:
Suggest rewriting this line and below as 

```markdown
## Usage

### Using the `T` within `Option<T>`

There are several common idioms for using the value which might be present in an `Option`:

> TODO: write a short explanatory paragraph about each of these

- `if let Some(t) = optional { ... }`. 
- `match optional { Some(t) => {...}, None => {...}}`
- `optional.map(|t| ...)` **Note**: operates on the contained value if present. The binding becomes an `Option<U>`, where the given closure is `FnOnce(T) -> U`.
- `optional.unwrap()`. **Important**: Unwrapping can ease development in the short term or in test code, but it introduces the possibility of runtime panics. It should be avoided when possible.

## Examples

// the examples already present
```

Reply:
Great, thanks! I am not familiar with using `optional.map(|t| ...)` so will read up on that later today to discuss usage.